uniffi/src/testing.rs

```diff
@@ -117,25 +122,82 @@ pub fn ensure_compiled_cdylib(pkg_dir: &str) -> Result<String> {
 ///
 /// Internally, this function does a bit of caching and concurrency management to avoid rebuilding
 /// the component for multiple testcases.
-pub fn ensure_generated_bindings<P1, P2, L>(cdylib_file: P1, out_dir: P2, language: L) -> Result<()>
+pub fn ensure_generated_bindings<P1, P2>(
+    idl_file: P1,
+    out_dir: P2,
+    language_ext: &str,
+) -> Result<()>
 where
     P1: AsRef<Path>,
     P2: AsRef<Path>,
-    L: TryInto<bindings::TargetLanguage, Error = anyhow::Error>,
 {
-    let cdylib_file: PathBuf = cdylib_file.as_ref().to_path_buf();
+    let idl_file: PathBuf = idl_file.as_ref().to_path_buf();
     let out_dir: PathBuf = out_dir.as_ref().to_path_buf();
-    let language: bindings::TargetLanguage = language.try_into()?;
     // Have we already generated these bindings?
     let mut generated_bindings = GENERATED_BINDINGS.lock().unwrap();
     // XXX TODO: figure out correct type of `HashSet` so that I don't need to `clone()` here...
-    if generated_bindings.contains(&(cdylib_file.clone(), out_dir.clone(), language)) {
+    if generated_bindings.contains(&(idl_file.clone(), out_dir.clone(), language_ext.to_owned())) {
         return Ok(());
     }
     // Nope, looks like we'll have to compile it afresh.
-    let ci = bindings::get_component_interface_from_cdylib(&cdylib_file)?;
-    bindings::write_bindings(&ci, &out_dir, language)?;
-    bindings::compile_bindings(&ci, &out_dir, language)?;
-    generated_bindings.insert((cdylib_file, out_dir, language));
+
+    let status = Command::new("uniffi-bindgen")
```

Comment:
We don't depend on `uniffi-bindgen`, but that's OK we call it on the command line :)

Reply:
I'm a little hesitant about this, since we don't have any guarantees that the `uniffi-bindgen` found on the system is the same version as used by the crate under test here. It seems like a footgun and potentially confusing experience for developers.

We have the `pkg_dir` here in order to compile the crate under test, I wonder if we should try to do something like:

* Inspect the cargo metadata to find the version of `uniffi` on which the crate depends.
* Shell out to cargo to build the matching version of `uniffi_bindgen` into our target dir, caching it similar to how we cache the compilation of the crate itself.
* Use that built version of the `uniffi-bindgen` command the run the tests.

I think this would resolve the pending CI failures, and also produce a more consistent/reliable developer experience.

(A simpler version of this might be to just build and use the `uniffi-bindgen` in the current workspace, which would suffice to get our CI passing).